internal/repo/datasource/sql_test.go

```diff
@@ -0,0 +1,357 @@
+				if !strings.Contains(err.Error(), "Unknown column") {
+					t.Errorf("Want: %v, Got: %v", "Unknown column", err)
+				}
+			},
+		},
+		{
+			name:   "FAILURE:: query error:: 2",
+			filter: map[string]interface{}{"userid": "1234"},
+			setupFunc: func() sqlDs {
+				db, mock, err := sqlmock.New()
+				if err != nil {
+					t.Fail()
+				}
+				dB := sqlDs{
+					sqlSvc: db,
+					table:  "newTemp",
+				}
+				mock.ExpectQuery(regexp.QuoteMeta("SELECT COUNT(`transaction_id`) FROM newTemp WHERE userid = '1234'")).WillReturnError(errors.New("query error")).WillReturnRows(sqlmock.NewRows([]string{"transaction_id"}).AddRow("1").AddRow("2").AddRow("3"))
+				mock.ExpectQuery("SELECT transaction_id, account_number, user_id, amount, transfer_to, created_at, updated_at, status, type, comment FROM newTemp WHERE user_id = '1234' ORDER BY created_at LIMIT 1 OFFSET 2 ;").WillReturnError(errors.New("Unknown column"))
+				return dB
+			},
+			validator: func(rows []model.Transaction, count int, err error) {
+				if !strings.Contains(err.Error(), "query error") {
+					t.Errorf("Want: %v, Got: %v", "query error", err)
+				}
+			},
+		},
+		{
+			name:   "FAILURE:: scan error:: 2",
+			filter: map[string]interface{}{"userid": "1234"},
+			setupFunc: func() sqlDs {
+				db, mock, err := sqlmock.New()
+				if err != nil {
+					t.Fail()
+				}
+				dB := sqlDs{
+					sqlSvc: db,
+					table:  "newTemp",
+				}
+				mock.ExpectQuery(regexp.QuoteMeta("SELECT COUNT(`transaction_id`) FROM newTemp WHERE userid = '1234'")).WillReturnRows(sqlmock.NewRows([]string{"transaction_id"}).AddRow("1"))
+				mock.ExpectQuery("SELECT transaction_id, account_number, user_id, amount, transfer_to, created_at, updated_at, status, type, comment FROM newTemp WHERE userid = '1234' ORDER BY created_at LIMIT 1 OFFSET 2 ;").WillReturnError(errors.New("query error"))
+				return dB
+			},
+			validator: func(rows []model.Transaction, count int, err error) {
+				if !strings.Contains(err.Error(), "query error") {
+					t.Errorf("Want: %v, Got: %v", "query error", err)
+				}
+			},
+		},
+	}
+
+	// to execute the tests in the table
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// STEP 1: seting up all instances for the specific test case
+			db := tt.setupFunc()
+			// STEP 2: call the test function
+			rows, count, err := db.Get(tt.filter, 1, 2)
+
+			// STEP 3: validation of output
+			if tt.validator != nil {
+				tt.validator(rows, count, err)
+			}
+
+			// STEP 4: clean up/remove up all instances for the specific test case
+			if tt.cleanupFunc != nil {
+				tt.cleanupFunc()
+			}
+		})
+	}
+}
+
+//
+func TestInsert(t *testing.T) {
+	// table driven tests
+	tests := []struct {
+		name        string
+		tableName   string
+		data        model.Transaction
+		setupFunc   func() (sqlDs, sqlmock.Sqlmock)
+		cleanupFunc func()
+		filter      map[string]interface{}
+		validator   func(sqlmock.Sqlmock, error)
+	}{
+		{
+			name: "SUCCESS:: Insert Article",
```

Comment:
Where is transaction converted to article?

Reply:
done